### PR TITLE
mon/MDSMonitor do not ignore mds's down:dne request

### PR DESCRIPTION
**Pull request by candychencan**
Fixes: https://tracker.ceph.com/issues/47881
Signed-off-by: chencan <chen.can2@zte.com.cn>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/mon/MDSMonitor.cc [hunk 1]
```diff
@@ -433,7 +433,7 @@ bool MDSMonitor::preprocess_beacon(MonOpRequestRef op)
     } else {
       effective_epoch = fsmap.get_filesystem(fscid)->mds_map.epoch;
     }
-    if (effective_epoch != m->get_last_epoch_seen()) {
+    if ((state != MDSMap::STATE_DNE) && (effective_epoch != m->get_last_epoch_seen())) {
```
Comment (batrick) [inline src/mon/MDSMonitor.cc:436]: Let's move this check under line 420. If the requested state is `STATE_DNE`, return false immediately.

**Review by batrick — CHANGES_REQUESTED** [review]

**Comment by batrick** [comment]
Please also update your commit title to the new PR title.

File: src/mon/MDSMonitor.cc (hunk 1 above)
Comment (candychencan) [reply src/mon/MDSMonitor.cc]: Fixed it ,thanks!

**Review by batrick — APPROVED** [review]

**Comment by batrick** [comment]
https://pulpito.ceph.com/?branch=wip-pdonnell-testing-20201021.232150

failures unrelated

**Pull request merged**